explorer/ast/expression.h

```diff
@@ -914,6 +914,17 @@ class WhereExpression : public RewritableMixin<Expression> {
   auto self_binding() const -> const GenericBinding& { return *self_binding_; }
   auto self_binding() -> GenericBinding& { return *self_binding_; }
 
+  auto enclosing_dot_self() const
+      -> std::optional<Nonnull<const GenericBinding*>> {
+    return enclosing_dot_self_;
+  }
+  // Sets the enclosing value of `.Self`. Can only be called once, by name
+  // resolution.
```

Comment:
As implemented, it can be called multiple times if the value is equal. Fix comment or change implementation?